### PR TITLE
fix mailto dialog

### LINKED ISSUE #3972: mailto: handler asks to type a contact recipient, but never finds a match
- Operating System: Windows
- Delta Chat Version: 1.46.1 (git: v1.33.0-881-g899f196d) (winstore install)
- Expected behavior: Popup appears with recent chats, filter by typing
- Actual behavior: Popup is empty, typing into filter does not suggest matches or otherwise allow user to proceed, even when typing literal addresses
- Steps to reproduce the problem: I've been using firefox's "email image" context menu. 
image

### PR DESCRIPTION
**Pull request by Simon-Laux**
closes #3972

### CODE REVIEW

File: src/renderer/components/dialogs/MailtoDialog/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.MailtoDialog {
+  height: 100%;
+}
+
+.MailtoDialogBody {
+  display: flex;
+  flex-direction: column;
+  padding-bottom: 0px;
```
Comment (farooqkz) [inline src/renderer/components/dialogs/MailtoDialog/styles.module.scss:8]: I think these should be in snake case to be consistent with other styles we've got?

File: src/renderer/components/dialogs/MailtoDialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import styles from './styles.module.scss'
+
+import type { DialogProps } from '../../../contexts/DialogContext'
+
+type Props = {
+  messageText: string
+}
+
+export default function MailtoDialog(props: Props & DialogProps) {
+  const { onClose, messageText } = props
```
Comment (farooqkz) [inline src/renderer/components/dialogs/MailtoDialog/index.tsx:24]:
```suggestion
export default function MailtoDialog({ onClose, messageText }: Props & DialogProps) {
```

File: src/renderer/components/dialogs/MailtoDialog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import { selectedAccountId } from '../../../ScreenController'
+import Dialog, { DialogBody, DialogContent, DialogHeader } from '../../Dialog'
+import useCreateDraftMessage from '../../../hooks/chat/useCreateDraftMesssage'
+import useTranslationFunction from '../../../hooks/useTranslationFunction'
+
+import styles from './styles.module.scss'
+
+import type { DialogProps } from '../../../contexts/DialogContext'
+
+type Props = {
```
Comment (farooqkz) [inline src/renderer/components/dialogs/MailtoDialog/index.tsx:19]: TS docs recommend preferring `interface` over `type` for defining object shapes. I am not requesting any change here. But let's get back to this later. This is a reminder.

File: src/renderer/components/dialogs/MailtoDialog/index.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  const { isChatLoaded, loadChats, chatCache } =
+    useLogicVirtualChatList(chatListIds)
+
+  const onChatClick = async (chatId: number) => {
+    createDraftMessage(accountId, chatId, messageText)
+    onClose()
+  }
+
+  const onSearchChange = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setQueryStr(e.target.value)
```
Comment (farooqkz) [inline src/renderer/components/dialogs/MailtoDialog/index.tsx:40]: What is the reason one is `async` and the other is not?

**Review by farooqkz — CHANGES_REQUESTED** [review]
Other than these "nitpicks", LGTM.

File: src/renderer/components/dialogs/MailtoDialog/index.tsx (hunk 4 above)
Comment (WofWca) [reply src/renderer/components/dialogs/MailtoDialog/index.tsx:40]: It was there before this MR

File: src/renderer/components/dialogs/MailtoDialog/index.tsx (hunk 3 above)
Comment (WofWca) [reply src/renderer/components/dialogs/MailtoDialog/index.tsx:19]: It was there before this MR

File: src/renderer/components/dialogs/MailtoDialog/index.tsx (hunk 2 above)
Comment (WofWca) [reply src/renderer/components/dialogs/MailtoDialog/index.tsx:24]: It was this way before this MR

**Pull request merged**